src/core.js

```diff
@@ -176,7 +176,10 @@ export function begin() {
 		runLoggingCallbacks( "begin", {
 			totalTests: Test.count,
 			modules: modulesLog
-		} ).then( unblockAndAdvanceQueue );
+		} ).then( unblockAndAdvanceQueue, function( err ) {
+			setTimeout( unblockAndAdvanceQueue );
```

Comment:
I've added tests for this scenario in https://github.com/qunitjs/qunit/pull/1634, and addressed the bug this fixes (https://github.com/qunitjs/qunit/issues/1446) via a different approach in https://github.com/qunitjs/qunit/pull/1638. In a nutshell, we handle this via the `unhandledrejection` event in the browser and Node.js, same as before, but previously that didn't work right due to it creating a fake test which we no longer do.